### PR TITLE
some console commands code refactor

### PR DESCRIPTION
**Pull request by nukeulater**

### CODE REVIEW

File: xlive/H2MOD/GUI/imgui_integration/Console/ImGui_ConsoleImpl.cpp [hunk 1]
```diff
@@ -144,12 +144,11 @@ int Console::TextEditCallback(ImGuiInputTextCallbackData* data)
...
 			memset(&console_data->m_completion_data->CompletionCandidate[i], 0, sizeof(ImGuiTextInputCompletionCandidate));
 			console_data->m_completion_data->CompletionCandidate[i].CompletionText = command->GetName();
-			console_data->m_completion_data->CompletionCandidate[i].CompletionVariable = nullptr;
 			console_data->m_completion_data->CompletionCandidate[i].CompletionDescription = command->GetDescription();
 
-			if (auto * var_command = dynamic_cast<ConsoleVarCommand*>(command))
+			if (command->CommandSetsVariable())
 			{
-				console_data->m_completion_data->CompletionCandidate[i].CompletionVariable = var_command->VarAsStr();
+				command->VarAsStr(console_data->m_completion_data->CompletionCandidate[i].CompletionVariable, 512);
```
Comment (Berthalamew) [inline xlive/H2MOD/GUI/imgui_integration/Console/ImGui_ConsoleImpl.cpp]: use NUMBEROF(CompletionVariable) instead of putting the size here

**Review by Berthalamew — CHANGES_REQUESTED** [review]

File: xlive/H2MOD/GUI/imgui_integration/Console/CommandHandler.h [hunk 2]
```diff
@@ -60,9 +67,18 @@ class ConsoleCommand
...
 	}
 
-    template<typename T = void*> T GetUserData() const 
-    { 
-        return (T)m_user_data; 
+    void SetCommandVarPtr(IComVar* varPtr)
+    {
+        if (varPtr != nullptr)
+        {
+            m_flags |= CommandFlag_SetsVariable;
```
Comment (Berthalamew) [inline xlive/H2MOD/GUI/imgui_integration/Console/CommandHandler.h:74]: m_flags should use the c_flags class instead of duplicating logic

**Review by Berthalamew — CHANGES_REQUESTED** [review]

File: xlive/H2MOD/GUI/imgui_integration/Console/CommandHandler.h (hunk 2 above)
Comment (nukeulater) [reply xlive/H2MOD/GUI/imgui_integration/Console/CommandHandler.h:74]: replacing functionality with h2 code is not in the scope of this PR

File: xlive/H2MOD/GUI/imgui_integration/Console/ImGui_ConsoleImpl.cpp (hunk 1 above)
Comment (nukeulater) [reply xlive/H2MOD/GUI/imgui_integration/Console/ImGui_ConsoleImpl.cpp]: done

File: xlive/H2MOD/GUI/imgui_integration/Console/CommandHandler.h (hunk 2 above)
Comment (Berthalamew) [reply xlive/H2MOD/GUI/imgui_integration/Console/CommandHandler.h:74]: That's fine with me.

**Review by Berthalamew — APPROVED** [review]

**Pull request merged**